src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/custom.py

```diff
@@ -39,18 +44,40 @@ def cli_redis_update_settings(client, resource_group_name, name, redis_configura
         )
     return client.create_or_update(resource_group_name, name, parameters=update_params)
 
-def cli_redis_create(client, resource_group_name, name, location, sku_name, # pylint:disable=too-many-arguments
-                     sku_family, sku_capacity, tags=None, redis_configuration=None,
+def cli_redis_update(instance, sku=None, vm_size=None):
+    if sku != None:
+        instance.sku.name = sku
+
+    if vm_size != None:
+        instance.sku.family = vm_size[0]
+        instance.sku.capacity = vm_size[1]
+
+    update_params = RedisCreateOrUpdateParameters(
+        instance.location,
+        instance.sku,
+        instance.tags,
+        instance.redis_version,
+        instance.redis_configuration,
+        instance.enable_non_ssl_port,
+        instance.tenant_settings,
+        instance.shard_count,
+        instance.subnet_id,
+        instance.static_ip,
+        )
+
+    return update_params
+
+def cli_redis_create(client, resource_group_name, name, location, sku, # pylint:disable=too-many-arguments
+                     vm_size, tags=None, redis_configuration=None,
                      enable_non_ssl_port=None, tenant_settings=None, shard_count=None,
                      subnet_id=None, static_ip=None):
     # pylint:disable=line-too-long
     """Create new Redis Cache instance
     :param resource_group_name: Name of resource group
     :param name: Name of redis cache
     :param location: Location
-    :param sku_name: What type of redis cache to deploy. Valid values: (Basic, Standard, Premium).
-    :param sku_family: Which family to use. Valid values: (C, P).
-    :param sku_capacity: What size of redis cache to deploy. Valid values for C family (0, 1, 2, 3, 4, 5, 6), for P family (1, 2, 3, 4)
+    :param sku: What type of redis cache to deploy. Valid values: (Basic, Standard, Premium).
```

Comment:
If you are using the **enum_choice_list, it will automatically list (and validate) the options.